packages/replay-next/src/contexts/TimelineContext.tsx

```diff
@@ -23,22 +29,32 @@ export type TimelineContextType = {
 export const TimelineContext = createContext<TimelineContextType>(null as any);
 
 export function TimelineContextRoot({ children }: PropsWithChildren<{}>) {
+  const client = useContext(ReplayClientContext);
   const [executionPoint, setExecutionPoint] = useState<ExecutionPoint>("0");
+  const [pauseId, setPauseId] = useState<PauseId | null>(null);
   const [time, setTime] = useState<number>(0);
 
   const [isPending, startTransition] = useTransition();
 
-  const update = useCallback((time: number, executionPoint: ExecutionPoint) => {
-    // Components might suspend in response to the this changing.
-    startTransition(() => {
-      setExecutionPoint(executionPoint);
-      setTime(time);
-    });
-  }, []);
+  const update = useCallback(
+    (time: number, executionPoint: ExecutionPoint) => {
+      // Components might suspend in response to the this changing.
+      startTransition(() => {
+        setExecutionPoint(executionPoint);
+        setTime(time);
+      });
+      async function updatePauseId() {
+        const pauseId = await pauseIdCache.readAsync(client, executionPoint, time);
+        setPauseId(pauseId);
```

Comment:
Components might suspend in response to the pause changing too, so it should probably also be set inside of the transition?

Reply:
(Or maybe a separate transition would be okay too?)

I'm a little worried about the "tearing" we might see by splitting these two updates apart.